eth/handler_eth.go

```diff
@@ -148,6 +152,26 @@ func (h *ethHandler) handleHeaders(peer *eth.Peer, headers []*types.Header) erro
 			}
 			peer.Log().Debug("Whitelist block verified", "number", headers[0].Number.Uint64(), "hash", want)
 		}
+
+		// fetch checkpoint from heimdall for whitelist
+		if headers[0].Number.Uint64()%h.chain.Config().Bor.Sprint == 0 && !h.chain.Engine().(*bor.Bor).WithoutHeimdall {
+			endBlockNum, endBlockHash, ok := h.whitelistCheckpoint()
+			if ok {
+				peer.Log().Debug("Whitelist block verified", "number", headers[0].Number.Uint64(), "hash", endBlockHash)
+				h.whitelist[endBlockNum] = endBlockHash
```

Comment:
Let's not pollute `h.whitelist` and use `h.whitelistCheckpoint` as different storage. `h.whitelist` has `config.whitelist`. Keeping them separate will be better.

Reply:
`h.whitelistCheckpoint` can store 10 checkpoints.